### PR TITLE
Fix SeekGE in DDB

### LINKED ISSUE #6558: [Bug]: Deleting imported directory results in ValidationException when listing parent directory
### What happened?

Steps to Reproduce:
1. I created a lakeFS repo (`oz-bug-1`), chose to add sample data
2. Removed branch protection rule from `main` branch
3. Imported a dataset from S3 (`s3://ds-vino/ozk/datasets/alpaca_training_imgs/`) on `main`, into `datasets/alpaca_training_imgs/`.
4. Data appears, everything looks good.
5. I then deleted the `datasets/` prefix on `main`
   ```shell
   $ lakectl fs rm --recursive lakefs://oz-bug-1/main/datasets/
   ```
6. Refresh the lakeFS objects screen, and I get this:

image

Interestingly, `lakectl fs ls lakefs://oz-bug-1/main/` does work!

### Expected behavior

Listing of `main/` with the `datasets/` prefix missing.

### lakeFS version

0.107.0

### How lakeFS is installed

lakeFS Cloud

### Affected clients

lakeFS UI

### Relevant log output

_No response_

### Contact details

_No response_

### PR DESCRIPTION
**Pull request by itaiad200**
closes #6558

### CODE REVIEW

File: pkg/kv/local/store_test.go [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package local_test
 
 import (
+	"context"
+	"github.com/treeverse/lakefs/pkg/kv"
```
Comment (nopcoder) [inline pkg/kv/local/store_test.go:5]: gofumpt is needed

File: pkg/kv/mem/store_test.go [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package mem_test
 
 import (
+	"context"
+	"github.com/treeverse/lakefs/pkg/kv"
```
Comment (nopcoder) [inline pkg/kv/mem/store_test.go:5]: imports format

File: pkg/kv/postgres/main_test.go [hunk 3]
```diff
@@ -49,7 +48,7 @@ func runDBInstance(dockerPool *dockertest.Pool) (string, func()) {
 	// create connection
 	var pgPool *pgxpool.Pool
 	port := resource.GetPort("5432/tcp")
-	uri := fmt.Sprintf("postgres://lakefs:lakefs@localhost:%s/lakefs_db?sslmode=disable", port)
+	uri := fmt.Sprintf("postgres://lakefs:lakefs@localhost:%s/%s?sslmode=disable", port, dbName)
```
Comment (nopcoder) [inline pkg/kv/postgres/main_test.go:51]: wrap dbName with call to url.PathEscape

File: pkg/kv/postgres/main_test.go
Comment (nopcoder) [inline pkg/kv/postgres/main_test.go:1]: Suggest to keep the same database name and have each test create a different db or scheme on the same instance.

File: pkg/kv/postgres/store_test.go [hunk 4]
```diff
@@ -1,6 +1,9 @@
 package postgres_test
 
 import (
+	"context"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/testutil"
```
Comment (nopcoder) [inline pkg/kv/postgres/store_test.go:6]: fmt

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/kv/postgres/store_test.go [hunk 5]
```diff
@@ -2,29 +2,45 @@ package postgres_test
...
+
 	kvtest.DriverTest(t, func(t testing.TB, ctx context.Context) kv.Store {
 		t.Helper()
-		databaseURI, cleanup := runDBInstance(pool, testutil.UniqueKVTableName())
+
+		conn, err := pgx.Connect(ctx, databaseURI)
+		if err != nil {
+			t.Fatalf("Unable to connect to database: %v", err)
+		}
+		defer conn.Close(context.Background())
```
Comment (nopcoder) [inline pkg/kv/postgres/store_test.go:27]:
```suggestion
		defer conn.Close(ctx)
```

File: pkg/kv/postgres/store_test.go [hunk 6]
```diff
@@ -2,29 +2,45 @@ package postgres_test
...
+
+		// create a new schema per test
+		schemaName := "test_schema" + testutil.UniqueName()
+		_, err = conn.Exec(context.Background(), fmt.Sprintf("CREATE SCHEMA IF NOT EXISTS %s;", schemaName))
```
Comment (nopcoder) [inline pkg/kv/postgres/store_test.go:31]: Use connection's `QuoteIdentifier` to quote the `schemaName` value.
Check the following if it works:
```suggestion
		_, err = conn.Exec(ctx, "CREATE SCHEMA IF NOT EXISTS " + conn. QuoteIdentifier(schemaName))
```

File: pkg/kv/postgres/store_test.go [hunk 7]
```diff
@@ -2,29 +2,45 @@ package postgres_test
...
+		if err != nil {
+			t.Fatalf("Error creating schema: %v", err)
```
Comment (nopcoder) [inline pkg/kv/postgres/store_test.go:33]:
```suggestion
			t.Fatalf("Error creating schema '%s': %s", schemaName, err)
```

File: pkg/kv/postgres/store_test.go [hunk 8]
```diff
@@ -2,29 +2,45 @@ package postgres_test
...
+		}
 
 		store, err := kv.Open(ctx, kvparams.Config{
 			Type:     postgres.DriverName,
-			Postgres: &kvparams.Postgres{ConnectionString: databaseURI, ScanPageSize: 10},
+			Postgres: &kvparams.Postgres{ConnectionString: fmt.Sprintf("%s&search_path=%s", databaseURI, schemaName), ScanPageSize: kvtest.MaxPageSize},
```
Comment (nopcoder) [inline pkg/kv/postgres/store_test.go:38]: Need to url parse the connection string to add qs parameter

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/kv/postgres/main_test.go (hunk 3 above)
Comment (itaiad200) [reply pkg/kv/postgres/main_test.go:51]: Done.

File: pkg/kv/postgres/main_test.go
Comment (itaiad200) [reply pkg/kv/postgres/main_test.go:1]: Done.

**Review by nopcoder — APPROVED** [review]
LGTM left some minor comments

File: pkg/kv/postgres/store_test.go (hunk 6 above)
Comment (itaiad200) [reply pkg/kv/postgres/store_test.go:31]: Can't. I simply escaped it

File: pkg/kv/postgres/store_test.go (hunk 7 above)
Comment (itaiad200) [reply pkg/kv/postgres/store_test.go:33]: Done.

**Pull request merged**